Microsoft.HealthVault.Fhir/FhirExtensions/DocumentReferenceExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Get Real Health.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation cdas (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.Thing;
+using static Hl7.Fhir.Model.DocumentReference;
+
+namespace Microsoft.HealthVault.Fhir.FhirExtensions
+{
+    public static class DocumentReferenceExtensions
+    {
+        public static ContentComponent GetFirstContentComponentWithData(this DocumentReference documentReference)
+        {
+            //We are considering only the first ContentComponent with data
+            var contentComponent = documentReference.Content.FirstOrDefault(o => o.Attachment != null && o.Attachment.Data != null);
```

Comment:
Let's change this to just be a First since there's a null check that throws an exception after every call to GetFirstContentComponentWithData

Reply:
Changed the statement to a First.